src/index.d.ts

```diff
@@ -1761,116 +1793,1703 @@ declare namespace admin.auth {
 }
 
 declare namespace admin.credential {
+
+  /**
+   * Interface that provides Google OAuth2 access tokens used to authenticate
+   * with Firebase services.
+   *
+   * In most cases, you will not need to implement this yourself and can instead
+   * use the default implementations provided by
+   * {@link admin.credential `admin.credential`}. 
+   */
   interface Credential {
+ 
+    /**
+     * Returns a Google OAuth2 access token object used to authenticate with
+     * Firebase services.
+     *
+     * This object contains the following properties:
+     * * `access_token` (`string`): The actual Google OAuth2 access token.
+     * * `expires_in` (`number`): The number of seconds from when the token was
+     *   issued that it expires.
+     *
+     * @return A Google OAuth2 access token object.
+     */
     getAccessToken(): Promise<admin.GoogleOAuthAccessToken>;
   }
-
+ 
+ 
+  /**
+   * Returns a credential created from the
+   * {@link
+    *    https://developers.google.com/identity/protocols/application-default-credentials
+    *    Google Application Default Credentials}
+    * that grants admin access to Firebase services. This credential can be used
+    * in the call to
+    * {@link
+    *   https://firebase.google.com/docs/reference/admin/node/admin#.initializeApp
+    *  `admin.initializeApp()`}.
+    *
+    * Google Application Default Credentials are available on any Google
+    * infrastructure, such as Google App Engine and Google Compute Engine.
+    *
+    * The Google Application Default Credential does not grant access to every
```

Comment:
This is no longer the case. Custom token creation works regardless of the credentials used. ID token verification also works most of the time, as long as there's a project ID available. Lets remove this entire paragraph.